src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -95,42 +80,57 @@ class OrderedRVD private(
 
   override def unpersist(): OrderedRVD = this
 
-  def orderedJoinDistinct(right: OrderedRVD, joinType: String): RDD[JoinedRegionValue] = {
+  def constrainToOrderedPartitioner(
+    ordType: OrderedRVDType,
+    newPartitioner: OrderedRVDPartitioner
+  ): OrderedRVD = {
+
+    require(ordType.rowType == typ.rowType)
+    require(ordType.kType isPrefixOf typ.kType)
+//    require(newPartitioner.kType isIsomorphicTo ordType.kType)
```

Comment:
It's frustrating because this should be an invariant satisfied by all `OrderedRVD`s. But I didn't see any simple way to make it pass, and I'm planning on a refactoring that removes `kType` from `OrderedRVDPartitioner`, getting rid of the redundancy. I'll just delete this.